tools/CheckH5.py

```diff
@@ -0,0 +1,207 @@
+                            or self.expected_h5_entity in check_h5)
+                        self.check_h5_file(check_h5, self.test_h5_entity,
+                                           self.expected_h5_entity)
+        self.unit_test.assertTrue(checks_performed)
+
+
+# The unit test object for performing all H5 checks for a given input file.
+class H5CheckTestCase(unittest.TestCase):
+
+    # The parameters for the H5 output checks are determined by the
+    # command-line arguments. The first argument (sys.argv[1]) is the yaml
+    # input file. The second argument is the directory in which to find the
+    # run's.h5 files.
+    #
+    # The H5 checks and arguments are parsed from the yaml file comments
+    # we require a format like:
+    # ```
+    # # H5Checks:
+    # #   - CheckLabel: "label"
+    # #     TestH5: "/h5_name.dat"
+    # #     AbsoluteTolerance: 1e-12
+    # #   - CheckLabel: "another_label"
+    # #     TestH5: "/h5_group_name"
+    # #     ExpectedH5: "/expected_h5_group_name"
+    # #     AbsoluteTolerance: 1e-11
+    # #     RelativeTolerance: 1e-6
+    # ```
+    # In particular, the entire comment block starting from `# H5Checks:` and
+    # ending where the indentation or comment breaks must be parsable as yaml
+    def test_h5_output(self):
+        input_filename = sys.argv[1]
+        run_directory = sys.argv[2]
+        with open(input_filename, 'r') as input_file:
+            lines = input_file.readlines()
+            line_index = 0
+            while line_index < len(
+                    lines) and lines[line_index] != "# H5Checks:\n":
+                line_index += 1
+            if line_index == len(lines):
+                raise RuntimeError(
+                    "Could not find '# H5Checks:' in input file. Please"
+                    " specify the h5 fields to check in the test yaml."
+                    " See tools/CheckH5.py for syntax details.")
+            h5_checks_lines = lines[line_index][2:]
+            line_index += 1
+            while lines[line_index][:6] == "#   - " or lines[
+                    line_index][:6] == "#     ":
+                h5_checks_lines += lines[line_index][2:]
+                line_index += 1
+        h5_check_list = []
+        parsed_yaml = yaml.safe_load(h5_checks_lines)
+        for check_block in parsed_yaml['H5Checks']:
+            print(check_block)
+            h5_check_list += [
+                H5Check(self, check_block.get("CheckLabel"),
+                        check_block.get("TestH5"),
+                        check_block.get("ExpectedH5"),
+                        check_block.get("AbsoluteTolerance"),
+                        check_block.get("RelativeTolerance"))
+            ]
+        for h5_check in h5_check_list:
+            # It is recommended to include subTest commands once python3 can be
+            # required:
+            # with self.subTest(test=h5_check.test_h5_label):
+            h5_check.perform_check(run_directory)
```

Comment:
Sure, I feel I frequently encounter cases in python where you can just drop a nested builtin data structure in, instead of writing a class. In a certain sense, any class with member variables can be made a dictionary...
However, I have found programming in that way frustrating in the end -- I prefer classes with the implied encapsulation (even if python doesn't seem to care much for encapsulation).
If you insist, I'll change it